source/logrepl/cdc_test.go

```diff
@@ -575,17 +662,12 @@ func TestCDCIterator_NextN(t *testing.T) {
 			VALUES (30, 'test-1', 100, false, 12.3, 14)`, table))
 		is.NoErr(err)
 
-		go func() {
-			time.Sleep(100 * time.Millisecond)
-			is.NoErr(i.Teardown(ctx))
-		}()
+		time.Sleep(100 * time.Millisecond)
+		is.NoErr(i.Teardown(ctx))
 
-		records, err := i.NextN(ctx, 5)
-		if err != nil {
-			is.True(strings.Contains(err.Error(), "logical replication error"))
-		} else {
-			is.True(len(records) > 0)
-		}
```

Comment:
When I read this part of the test, I wasn't sure what the expected behavior was (i.e., whether we expected an error or not). Based on the test description ("subscription termination"), it appeared to me that an error is what we should always expect.